# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [40]:
import pandas as pd
import numpy as np
from scipy import stats
pd.set_option('display.max_columns', None)
import math

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

***
### Question 1: What test is appropriate for this problem? Does CLT apply?

The question here is find out if there is a difference in the average * call rate * between White race applicants and other race applicants. So, we are comparing the mean of two samples, $X bar$, where the population standard deviation &sigma; is not known. The sample size, $n$ is > 30. Considering all these, we will use a two sample $t$ test to solve this. Yes, Central Limit Theorem does apply. 

***
### Question 2:    What are the null and alternate hypotheses? ###

$H$<sub>o</sub> : $Xbar$<sub>white pop</sub> = $Xbar$<sub>black pop</sub>

$H$<sub>a</sub> : $Xbar$<sub>white pop</sub> &ne; $Xbar$<sub>black pop</sub>


***
### Question 3: Compute margin of error, confidence interval, and p-value

In [23]:
data_b=data[data.race=='b']

In [29]:
data_b.head(5)

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
7  b  1          3       4        21       0          1         0         1   
8  b  1          4       3         3       0          0         0         0   
9  b  1          4       2         6       0          1         0         0   

   occupspecific  occupbroad  workinschool  email  computerskills  \
2             19           1             1      0               1   
3            313           5             0      1               1   
7            313           5             0      1               1   
8            316           6             0      0               1   
9            263           4             1      1               0   

   specialskills firstname sex race    h    l  call city kind   adid  \
2              0   Lakisha   f    b  0.0  1.0   0.0    c    a  384.0   
3              1   Latonya   f    b  1.0  0.0   0.0    c    a  384.0   
7              1     Kenya   f    b  1.0  0.0   0.0    c    a  385.0   
8              1   Latonya   f    b  0.0  1.0   0.0    c    s  386.0   
9              1    Tyrone   m    b  1.0  0.0   0.0    c    s  386.0   

   fracblack  fracwhite  lmedhhinc  fracdropout  fraccolp       linc  col  \
2   0.104301   0.837370  10.466754     0.101335  0.591695  10.540329  1.0   
3   0.336165   0.637370  10.431908     0.108848  0.406576  10.412141  0.0   
7   0.116624   0.728339  10.287047     0.139843  0.365636   9.933725  0.0   
8   0.080736   0.888374  10.408828     0.233687  0.087285   9.532859  1.0   
9   0.992043   0.004814   8.841738     0.295093  0.053182   8.507345  1.0   

  expminreq schoolreq  eoe  parent_sales  parent_emp  branch_sales  \
2         5            1.0           NaN         NaN           NaN   
3         5            1.0           NaN         NaN           NaN   
7      some            1.0           9.4       143.0           9.4   
8                      1.0          40.0       135.0          40.0   
9                      1.0          40.0       135.0          40.0   

   branch_emp  fed  fracblack_empzip  fracwhite_empzip  lmedhhinc_empzip  \
2         NaN  NaN               NaN               NaN               NaN   
3         NaN  NaN               NaN               NaN               NaN   
7       143.0  0.0          0.204764          0.727046         10.619399   
8       135.0  0.0          0.008141          0.973413         11.137956   
9       135.0  0.0          0.008141          0.973413         11.137956   

   fracdropout_empzip  fraccolp_empzip  linc_empzip  manager  supervisor  \
2                 NaN              NaN          NaN      0.0         1.0   
3                 NaN              NaN          NaN      0.0         1.0   
7            0.070493         0.369903    10.007352      0.0         0.0   
8            0.047958         0.413306    10.393723      0.0         0.0   
9            0.047958         0.413306    10.393723      0.0         0.0   

   secretary  offsupport  salesrep  retailsales  req  expreq  comreq  educreq  \
2        0.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
3        0.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
7        1.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
8        0.0         0.0       1.0          0.0  0.0     0.0     0.0      0.0   
9        0.0         0.0       1.0          0.0  0.0     0.0     0.0      0.0   

   compreq  orgreq  manuf  transcom  bankreal  trade  busservice  othservice  \
2      1.0     0.0    1.0       0.0       0.0    0.0         0.0         0.0   
3      1.0     0.0    1.0       0.0       0.0    0.0         0.0         0.0   
7      1.0     1.0    0.0       0.0       0.0    0.0         0.0         1.0   
8      0.0     0.0    0.0       0.0       0.0    1.0         0.0         0

In [31]:
len(data_b)

2435

In [25]:
data_w=data[data.race=='w']

In [28]:
data_w.head(5)

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
4  b  1          3       3        22       0          0         0         0   
5  b  1          4       2         6       1          0         0         0   
6  b  1          4       2         5       0          1         0         0   

   occupspecific  occupbroad  workinschool  email  computerskills  \
0             17           1             0      0               1   
1            316           6             1      1               1   
4            313           5             1      1               1   
5            266           4             0      0               0   
6             13           1             1      1               1   

   specialskills firstname sex race    h    l  call city kind   adid  \
0              0   Allison   f    w  0.0  1.0   0.0    c    a  384.0   
1              0   Kristen   f    w  1.0  0.0   0.0    c    a  384.0   
4              0    Carrie   f    w  1.0  0.0   0.0    c    a  385.0   
5              1       Jay   m    w  0.0  1.0   0.0    c    s  386.0   
6              1      Jill   f    w  1.0  0.0   0.0    c    s  386.0   

   fracblack  fracwhite  lmedhhinc  fracdropout  fraccolp       linc  col  \
0   0.989360   0.005500   9.527484     0.274151  0.037662   8.706325  1.0   
1   0.080736   0.888374  10.408828     0.233687  0.087285   9.532859  0.0   
4   0.397595   0.180196   9.876219     0.312873  0.030847   8.728264  0.0   
5   0.336165   0.637370  10.431908     0.108848  0.406576  10.412141  1.0   
6   0.002796   0.952140  10.453601     0.236445  0.124980   9.621058  1.0   

  expminreq schoolreq  eoe  parent_sales  parent_emp  branch_sales  \
0         5            1.0           NaN         NaN           NaN   
1         5            1.0           NaN         NaN           NaN   
4      some            1.0           9.4       143.0           9.4   
5                      1.0          40.0       135.0          40.0   
6                      1.0          40.0       135.0          40.0   

   branch_emp  fed  fracblack_empzip  fracwhite_empzip  lmedhhinc_empzip  \
0         NaN  NaN               NaN               NaN               NaN   
1         NaN  NaN               NaN               NaN               NaN   
4       143.0  0.0          0.204764          0.727046         10.619399   
5       135.0  0.0          0.008141          0.973413         11.137956   
6       135.0  0.0          0.008141          0.973413         11.137956   

   fracdropout_empzip  fraccolp_empzip  linc_empzip  manager  supervisor  \
0                 NaN              NaN          NaN      0.0         1.0   
1                 NaN              NaN          NaN      0.0         1.0   
4            0.070493         0.369903    10.007352      0.0         0.0   
5            0.047958         0.413306    10.393723      0.0         0.0   
6            0.047958         0.413306    10.393723      0.0         0.0   

   secretary  offsupport  salesrep  retailsales  req  expreq  comreq  educreq  \
0        0.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
1        0.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
4        1.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
5        0.0         0.0       1.0          0.0  0.0     0.0     0.0      0.0   
6        0.0         0.0       1.0          0.0  0.0     0.0     0.0      0.0   

   compreq  orgreq  manuf  transcom  bankreal  trade  busservice  othservice  \
0      1.0     0.0    1.0       0.0       0.0    0.0         0.0         0.0   
1      1.0     0.0    1.0       0.0       0.0    0.0         0.0         0.0   
4      1.0     1.0    0.0       0.0       0.0    0.0         0.0         1.0   
5      0.0     0.0    0.0       0.0       0.0    1.0         0.0         0

In [30]:
len(data_w)

2435

In [32]:
b_calls = data_b.call

In [35]:
w_calls = data_w.call

In [37]:
#mean of both samples and their difference - (0.09650924056768417, 0.0644763857126236, 0.03203285485506058)
w_calls.mean(),b_calls.mean(),w_calls.mean()-b_calls.mean()

(0.09650924056768417, 0.0644763857126236, 0.03203285485506058)

In [39]:
#standard deviation of both samples - (0.2953455150127411, 0.24564945697784424)
w_calls.std(),b_calls.std()

(0.2953455150127411, 0.24564945697784424)

_ margin of error =  $t$<sub>critical</sub> $X$ $Standard Error$ _

_Standard Error = &radic;  ( (variance<sub>w</sub> / n<sub>w</sub>) + (variance<sub>b</sub> / n<sub>b</sub> ] _

The Degrees of Freedom for this is 4711

The $t$<sub>critical</sub> value is 1.960

In [43]:
standard_error = math.sqrt(w_calls.var()/len(w_calls) + b_calls.var()/len(b_calls))

In [45]:
standard_error * 1.960

0.015258417380692

confidence_interval = +- 0.01526, i.e. any difference above or below this interval will lead to the rejection of $H$<sub>o</sub>. In this case the difference in mean between the two samples is 0.032, so $H$<sub>o</sub> can be rejected

In [49]:
0.03203285485506058/ standard_error

4.1147383735462695

In [48]:
stats.ttest_ind(w_calls,b_calls)

Ttest_indResult(statistic=4.1147052908617514, pvalue=3.9408021031288859e-05)

** The $p$ value is 3.9408021031288859e-05 **